### ISSUE TITLE
维度不匹配

### ISSUE THREAD

**Issue by a1394156524** [zhengchen1999/DAT#22]
![微信图片_20231215101939](https://github.com/zhengchen1999/DAT/assets/118806504/10517f9b-1538-49aa-9583-b958b3720d25)
您好，我训练用您默认的参数，训练DATx2和DATx4的都没有问题，但是训练x3时出现以上的错误，请问该怎么解决？

**Comment by zhengchen1999** [comment]
I'm sorry for the late reply.
The issue was due to my mistake: an error in x3 training config file ([here](https://github.com/zhengchen1999/DAT/blob/main/options/Train/train_DAT_x3.yml#L44)). I have already corrected it.

**Comment by orangegirll** [comment]
Excuse me, the following questions always arise when training, regardless of which profile is taken
ValueError: Expected more than 1 value per channel when training, got input size torch.Size([1, 22, 1, 1])
![微信图片_20240112231842](https://github.com/zhengchen1999/DAT/assets/108671480/aa65bd77-79f3-4e94-87f2-6f67b247f7b1)

**Comment by orangegirll** [comment]
Excuse me, the following questions always arise when training, regardless of which profile is taken
ValueError: Expected more than 1 value per channel when training, got input size torch.Size([1, 22, 1, 1])
![Uploading 微信图片_20240112232359.png…]()

**Comment by orangegirll** [comment]
Excuse me, the following questions always arise when training, regardless of which profile is taken
ValueError: Expected more than 1 value per channel when training, got input size torch.Size([1, 22, 1, 1])
![微信图片_20240112232359](https://github.com/zhengchen1999/DAT/assets/108671480/049a015b-eaf7-4ad1-a948-279cb2857af9)

**Comment by zhengchen1999** [comment]
What is the size of your input? It seems here that it is caused by incorrect input size."

**Comment by Chenweihai666** [comment]
Hello, while I was debugging, I also encountered the same problem. The input is (1, 3, 64, 64)

**Comment by Chenweihai666** [comment]
Because BatchNorm2d requires the input of batch_size greater than 1

**Comment by zhengchen1999** [comment]
You're right. If the input batch size is 1, then you should set model.eval().
